internal/pkg/diagnostics/dianostics_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package diagnostics
+
+import (
+	"archive/zip"
+	"bytes"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+)
+
+func TestZipLogs(t *testing.T) {
+	// Setup a directory structure of: logs/httpjson/log.ndjson
+	{
+		paths.SetTop(t.TempDir())
+		dir := filepath.Join(paths.Home(), "logs/sub-dir")
+		require.NoError(t, os.MkdirAll(dir, 0o700))
+		require.NoError(t, os.WriteFile(filepath.Join(dir, "log.ndjson"), []byte(".\n"), 0o600))
+	}
+
+	// Zip the logs directory.
+	buf := new(bytes.Buffer)
+	w := zip.NewWriter(buf)
+	require.NoError(t, zipLogs(w, time.Now()))
+	require.NoError(t, w.Close())
+
+	type zippedItem struct {
+		Name  string
+		IsDir bool
+	}
+
+	// Read back the contents.
+	r, err := zip.NewReader(bytes.NewReader(buf.Bytes()), int64(buf.Len()))
+	require.NoError(t, err)
+	var observed []zippedItem
+	for _, f := range r.File {
+		observed = append(observed, zippedItem{Name: f.Name, IsDir: f.FileInfo().IsDir()})
+	}
+
+	// Verify the results.
+	expected := []zippedItem{
+		{"logs/", true},
```

Comment:
@michalpristas I pulled in the latest code and this test points out a problem with the newly added `elastic-agent-{commit}` directory in that it is missing a directory entry within the Zip file. 

Depending on how robust a particular unzip tool used is, this can lead to problems when unzipping. Sometimes a tool will try to write files into a directory that doesn't exist and fail (or worse, silently ignore the problem and not write any of the files under that dir).

The test is passing now, so this could be followed up in a separate change.